src/main/scala/is/hail/distributedmatrix/RowMatrix.scala

```diff
@@ -40,6 +82,20 @@ class RowMatrix(val hc: HailContext,
   // length nPartitions + 1, first element 0, last element rdd2 count
   def partitionStarts(): Array[Long] = partitionCounts().scanLeft(0L)(_ + _)
     
+  def toLocalMatrix(): DenseMatrix[Double] = {
+    // if _nRows is defined, check that it's a valid Int
+    require(_nRows.forall(_ <= Int.MaxValue), "The number of rows of this matrix should be less than or equal to " +
+        s"Int.MaxValue. Currently numRows: ${ _nRows.get }")
+    
+    val a = rows.map(_._2).collect() // fails if nRows > Int.MaxValue
```

Comment:
I don't think the comment is necessary

Reply:
If `_nRows` isn't set we get a bad error message here. Under what circumstances is `_nRows` not set? I'm inclined to require that we know the number of rows, even if this means counting them when converting from Spark. I don't think we'll be doing that much any more anyway.

Reply:
I'm more worried about extra actions after conversion from the Hail side, not from Spark: if required, we'd also need to count when converting from MatrixTable to RowMatrix or when filtering a RowMatrix (functionality to come). For now, I'm mirroring the fact that partitionCounts is optional in MatrixTable (available right after a read), and Cotton intends the counts to become non-optional and propagated through filters etc at which point I'll change RowMatrix to work similarly.

Reply:
OK